xbmc/cores/dvdplayer/DVDCodecs/Overlay/DVDOverlayCodecFFmpeg.cpp

```diff
@@ -237,10 +239,28 @@ CDVDOverlay* CDVDOverlayCodecFFmpeg::GetOverlay()
     if(m_Subtitle.rects[m_SubtitleIndex] == NULL)
       return NULL;
 
-    AVSubtitleRect& rect = *m_Subtitle.rects[m_SubtitleIndex];
+    AVSubtitleRect rect = *m_Subtitle.rects[m_SubtitleIndex];
     if (rect.pict.data[0] == NULL)
       return NULL;
 
+    m_height = m_pCodecContext->height;
+    m_width  = m_pCodecContext->width;
+
+    RENDER_STEREO_MODE render_stereo_mode = g_graphicsContext.GetStereoMode();
+    if (render_stereo_mode != RENDER_STEREO_MODE_OFF)
```

Comment:
@koying this makes subtitles appear to the right. Without this block they look ok. See forum:
http://forum.kodi.tv/showthread.php?tid=244009
On my system the first couple of subtitles look ok because gfx context has not switched to stereo mode. Once the switch happened, width is set to 960 and subs are rendered incorrectly.